core/sr-primitives/src/traits.rs

```diff
@@ -788,8 +788,8 @@ pub enum DispatchError {
 	/// General error to do with the transaction's proofs (e.g. signature).
 	BadProof,
 
-/*	/// General error to do with actually executing the dispatched logic.
-	User(&'static str),*/
+	/// General error to do with a limit being reached that may be reset on the next block.
+	BlockExhausted,
```

Comment:
Then maybe get rid of the `Resource` that I added above? they seem to have an overlap (at least when I added resource sth like _block exhausted_ was also a candidate for the same purpose).